File: get-agent-card.py
#!/usr/bin/env python3
"""
A2A HelloWorld Agent Card Fetcher
Simple Python script to get agent card information
"""

import requests
import json
import sys

def get_agent_card(base_url="https://a2a-helloworld-1dd6ef1d53ae.herokuapp.com"):
    """Get the agent card from the deployed A2A agent"""
    try:
        response = requests.get(f"{base_url}/agent/card", timeout=10)
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"❌ Error fetching agent card: {e}")
        return None

def get_agent_health(base_url="https://a2a-helloworld-1dd6ef1d53ae.herokuapp.com"):
    """Get the agent health status"""
    try:
        response = requests.get(f"{base_url}/agent/health", timeout=10)
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"❌ Error fetching agent health: {e}")
        return None

def call_jsonrpc_method(method, params=None, base_url="https://a2a-helloworld-1dd6ef1d53ae.herokuapp.com"):
    """Call a JSON-RPC method on the agent"""
    try:
        payload = {
            "jsonrpc": "2.0",
            "method": method,
            "params": params or {},
            "id": 1
        }
        response = requests.post(
            f"{base_url}/jsonrpc",
            json=payload,
            headers={"Content-Type": "application/json"},
            timeout=10
        )
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"❌ Error calling JSON-RPC method '{method}': {e}")
        return None

def main():
    print("🤖 A2A HelloWorld Agent Card Fetcher")
    print("=" * 40)
    
    # Get agent card
    print("📋 Fetching agent card...")
    agent_card = get_agent_card()
    if agent_card:
        print("✅ Agent Card:")
        print(json.dumps(agent_card, indent=2))
    else:
        print("❌ Failed to get agent card")
        sys.exit(1)
    
    print("\n" + "=" * 40)
    
    # Get health status
    print("💚 Checking agent health...")
    health = get_agent_health()
    if health:
        print("✅ Health Status:")
        print(json.dumps(health, indent=2))
    
    print("\n" + "=" * 40)
    
    # Test JSON-RPC greeting
    print("👋 Testing JSON-RPC greeting...")
    greeting = call_jsonrpc_method("greeting")
    if greeting:
        print("✅ Greeting Response:")
        print(json.dumps(greeting, indent=2))
    
    print("\n✅ All tests completed successfully!")

if __name__ == "__main__":
    main()

File: get-agent-card.js
#!/usr/bin/env node
/**
 * A2A HelloWorld Agent Card Fetcher
 * Simple Node.js script to get agent card information
 */

const https = require('https');

const AGENT_URL = 'https://a2a-helloworld-1dd6ef1d53ae.herokuapp.com';

function makeRequest(path, method = 'GET', data = null) {
    return new Promise((resolve, reject) => {
        const options = {
            hostname: 'a2a-helloworld-1dd6ef1d53ae.herokuapp.com',
            port: 443,
            path: path,
            method: method,
            headers: {
                'Content-Type': 'application/json',
                'User-Agent': 'A2A-Agent-Tester/1.0'
            }
        };

        const req = https.request(options, (res) => {
            let body = '';
            res.on('data', (chunk) => {
                body += chunk;
            });
            res.on('end', () => {
                try {
                    const jsonData = JSON.parse(body);
                    resolve(jsonData);
                } catch (e) {
                    resolve(body);
                }
            });
        });

        req.on('error', (err) => {
            reject(err);
        });

        if (data) {
            req.write(JSON.stringify(data));
        }
        req.end();
    });
}

async function getAgentCard() {
    try {
        console.log('📋 Fetching agent card...');
        const agentCard = await makeRequest('/agent/card');
        console.log('✅ Agent Card:');
        console.log(JSON.stringify(agentCard, null, 2));
        return agentCard;
    } catch (error) {
        console.error('❌ Error fetching agent card:', error.message);
        return null;
    }
}

async function getAgentHealth() {
    try {
        console.log('💚 Checking agent health...');
        const health = await makeRequest('/agent/health');
        console.log('✅ Health Status:');
        console.log(JSON.stringify(health, null, 2));
        return health;
    } catch (error) {
        console.error('❌ Error fetching agent health:', error.message);
        return null;
    }
}

async function testJsonRpcGreeting() {
    try {
        console.log('👋 Testing JSON-RPC greeting...');
        const greeting = await makeRequest('/jsonrpc', 'POST', {
            jsonrpc: '2.0',
            method: 'greeting',
            params: {},
            id: 1
        });
        console.log('✅ Greeting Response:');
        console.log(JSON.stringify(greeting, null, 2));
        return greeting;
    } catch (error) {
        console.error('❌ Error calling JSON-RPC greeting:', error.message);
        return null;
    }
}

async function main() {
    console.log('🤖 A2A HelloWorld Agent Card Fetcher');
    console.log('='.repeat(40));
    
    const agentCard = await getAgentCard();
    if (!agentCard) {
        process.exit(1);
    }
    
    console.log('\n' + '='.repeat(40));
    await getAgentHealth();
    
    console.log('\n' + '='.repeat(40));
    await testJsonRpcGreeting();
    
    console.log('\n✅ All tests completed successfully!');
}

main().catch(console.error);
